### ISSUE TITLE
Feature Request - RFC 3548 Base64 encoding of filenames

### ISSUE THREAD

**Issue by ScottRFrost** [stefankueng/CryptSync#1]
I believe you could save a lot by encoding the filenames in RFC 3548 filename safe Base64 (/ becomes _, + becomes -, remove =) instead of writing them out as hex, which is taking a lot of space.   It would still make the file names a little bit larger, but not nearly where they are today with hex.

Original Filename - 50 Characters:
WeifenLuo.WinFormsUI.Docking.ThemeVS2012Light.dll

Hex - 103 characters (106% larger):
fe...e7.7z

Base64 with RFC 3548 substitutes - 68 characters (36% larger):
_n...uc

I'm a C# dev and I don't know much C++, but I think in ```FolderSync.cpp - std::wstring CFolderSync::GetEncryptedFilename``` just before you do ```if (useGPG)``` you'd just have to convert encryptFIlename to base64, do the 3 substitutions, then you're all set.  When decrypting, all you'd have to do is do the substitutions in reverse to get the regular base64, convert that to hex, decrypt as usual.

**Comment by ScottRFrost** [comment]
Link to RFC - https://tools.ietf.org/html/rfc3548#section-4

TLDR:
/  ==>  _  (underscore)
+  ==> -  (dash or minus)
=  ==>  Removed
